pandas/core/groupby/ops.py

```diff
@@ -1243,13 +1245,7 @@ def _chop(self, sdata: Series, slice_obj: slice) -> Series:
         # fastpath equivalent to `sdata.iloc[slice_obj]`
         mgr = sdata._mgr.get_slice(slice_obj)
         # __finalize__ not called here, must be applied by caller if applicable
-
-        # fastpath equivalent to:
-        # `return sdata._constructor(mgr, name=sdata.name, fastpath=True)`
-        obj = type(sdata)._from_mgr(mgr)
-        object.__setattr__(obj, "_flags", sdata._flags)
```

Comment:
are these still getting pinned?

Reply:
`__finalize__` should take care of that (there is a `test_finalize.py` test that was changed in the diff, which was testing that this was not yet implemented. Although it was testing it with `attrs` and not with `flags`, but both should be handled by finalize).